src/schemas/Polymorphic.js

```diff
@@ -41,10 +43,11 @@ export default class PolymorphicSchema {
   }
 
   denormalizeValue(value, unvisit) {
-    if (!this.isSingleSchema && !value.schema) {
+    const mutableValue = isImmutable(value) ? value.toJS() : value;
```

Comment:
Yea it can be expensive. From [the docs](https://facebook.github.io/immutable-js/docs/#/Map/toJS) for `.toJS()`:
> Deeply converts this Keyed collection to equivalent native JavaScript Object.

Since we only care about accessing the `schema` and `id` key here I think this would be better written as:
```js
const schemaKey = isImmutable(value) ? value.get('schema') : value.schema;
if (!this.isSingleSchema && !schemaKey) {
  return value;
}
const schema = this.isSingleSchema ? this.schema : this.schema[schemaKey];
const id = isImmutable(value) ? value.get('id') : value.id;
return unvisit(id || value, schema);
```